lib/gyro/template.rb

```diff
@@ -12,24 +12,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+require 'yaml'
+
 module Gyro
   # Gyro Template Helper
   #
   module Template
+
+    # Print template list representation
+    #
     def self.print_list
-      Gyro::Template.directory.children.sort_by(&:basename).each do |entry|
+      template_config_path = Gyro::Template.directory + 'config.yml'
+      template_config = YAML.load_file(template_config_path)
+      alias_template_hash = template_config['alias']
+      deprecated_template_list = template_config['deprecated']
+
+      alias_template_list = alias_template_hash.map{ |key, value| key }.to_a
+      directory_template_list = Gyro::Template.directory.children.select(&:directory?).map { |element| element.basename.to_s }
```

Comment:
I wonder if `rubocop` will encourage you to use `.map(&:basename).map(&:to_s)` instead?… (No personal preference here, but that alternative syntax can be useful to easier split in multiple lines if rubocop is unhappy)

Reply:
Rubocop don't blame me for this but if I use `.select(&:directory?)` before, in my opinion it's better to keep the same syntax for the following `map`. So I change and I implement your solution.